### PR TITLE
new doc page first draft

### PR DESCRIPTION
**Pull request by 4c74356b41**
howdy, first draft of azure python aks docs and fixed the other azure doc a bit, it had couple silly mistakes

### CODE REVIEW

**Comment by 4c74356b41** [comment]
I did test the page locally, it appears to be fine.

File: quickstart/azure/tutorial-azure-kubernetes-service.md [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+        + 11 created
+
+    Duration: 7m30s
+
+    Permalink: https://app.pulumi.com/4c74356b41/aks/dev/updates/10
+    ```
+
+    To see the full details of the deployment and the resources that are now part of the stack, open the update permalink in a browser.
+
+5.  To view the provisioned resources on the command line, run `pulumi stack`. You'll also see a [stack output](/reference/stack.html#output) corresponding to the kubernetes config file you can use to access the AKS cluster we created.
```
Comment (metral) [inline quickstart/azure/tutorial-azure-kubernetes-service.md]: Let's omit this section, as its extraneous.

Comment (4c74356b41) [reply quickstart/azure/tutorial-azure-kubernetes-service.md]: section 5?

Comment (metral) [reply quickstart/azure/tutorial-azure-kubernetes-service.md]: Yes please :)

Comment (4c74356b41) [reply quickstart/azure/tutorial-azure-kubernetes-service.md]: done

**Comment by 4c74356b41** [comment]
@metral

**Comment by metral** [comment]
Thanks for removing the requested section. Any plans to create parity of this doc to the wording and styling of the EKS and GKE docs listed?

> Given that the EKS and GKE follow almost identical steps and have similar content, it'd be great if we can modify this PR to mirror those tutorials as close as possible to create parity for users across all platforms.
> 
> Here's the [EKS](https://github.com/pulumi/docs/blob/master/quickstart/aws/tutorial-eks.md) and the [GKE](https://github.com/pulumi/docs/blob/master/quickstart/gcp/tutorial-gke.md) tutorials to reference.

**Comment by 4c74356b41** [comment]
with all due respect, can you be more precise (i cant read your mind, you know)? except for the creating service\deployment these are identical. and I cant create anything from yaml, because python. @

**Comment by metral** [comment]
> can you be more precise

Sure. Could you please do the following:

- Do not list numbered lists using actual numbers e.g. `1...2...3` - Markdown allows you to number them all `1.` and at render time it will assemble the order automatically for us. See the EKS raw [example](https://raw.githubusercontent.com/pulumi/docs/master/quickstart/aws/tutorial-eks.md), and how all of the numbered items begin with `1.  `

- Move the pre-reqs section to a separate file in the `_includes` folder for future reusability, and then include it in this doc using: `{% include aks-prereqs.md %}`, assuming the file is named `aks-prereqs.md`
- In section 1 - "In a new folder aks.."
  - break up the first two commands (`pulumi new`, `cd aks` ) into separate sections, each with the header text as in https://github.com/pulumi/docs/blob/master/quickstart/aws/tutorial-eks.md#create-a-new-eks-cluster-new-eks-cluster
  - for `pip install`, please also include the header text in the link above
- In section 2 - "Open __main__.py ...`
  - please include the header text detailing what the user is about to do before the code block - just as in the EKS link above.
  e.g.

> The index.ts occupies the role as the main entrypoint in our Pulumi program. In it, we are going to declare:
>
> - The resources we want in Azure to provision the AKS cluster based on our cluster configuration settings,
> - The kubeconfig file to access the cluster, and
> - The initialization of a Pulumi Kubernetes provider with the kubeconfig, so that we can deploy Kubernetes resources to the cluster once its ready in the next steps.

- Merge section 3 and 4, and use the same header text as in EKS "To preview and deploy changes..." as in the EKS link above.

>  To preview and deploy changes, run pulumi update and select "yes."
> 
> The update sub-command shows a preview of the resources that will be created and prompts on whether to proceed with the deployment. Note that the stack itself is counted as a resource, though it does not correspond to a physical cloud resource.
> 
> You can also run pulumi up --diff to see and inspect the diffs of the overall changes expected to take place.
> 
> Running pulumi up will deploy the AKS cluster. Note, provisioning a new EKS cluster takes between X minutes.

- Include the section to use pulumi providers with the exact content and example listed in steps 1 and 2 of that [EKS section](https://github.com/pulumi/docs/blob/master/quickstart/aws/tutorial-eks.md#access-the-kubernetes-cluster-using-pulumi-providers)
- Add the missing [Access the Kubernetes Cluster using kubectl](https://github.com/pulumi/docs/blob/master/quickstart/aws/tutorial-eks.md#access-the-kubernetes-cluster-using-pulumi-providers) section.
- Add the missing [Experimentation](https://github.com/pulumi/docs/blob/master/quickstart/aws/tutorial-eks.md#experimentation) section.
- In the summary section, mirror the [EKS summary](https://github.com/pulumi/docs/blob/master/quickstart/aws/tutorial-eks.md#summary) section, with follow up steps and embedded youtube video, and drop the duplicate line "To clean up resources, run pulumi destroy" as we already call that out in the previous "Clean up" section

Let me know if you have any further questions

**Comment by 4c74356b41** [comment]
1. that is intentional, i dont like it when its all at 1.
2-x. like I said, you want them to look the same, ok. that is easy.
kubectl section isnt missing, its just a bit shorter (again, you want them to look the same)
experimentation is not possible with python
not clear about summary. you want me to literally copy-paste it or something else? I assume there is not joe beda python video
not sure how to resolve that conflict, that change is intentional, since the original contains many errors

**Comment by 4c74356b41** [comment]
@metral

**Comment by metral** [comment]
I do not see any new commits to reflect the changes [requested](https://github.com/pulumi/docs/pull/897#issuecomment-468429696).

We would like this doc have complete parity with the EKS and GKE referenced above. Please consider addressing the requested items listed that are capable of being implemented to move this forward.

Thanks

**Comment by 4c74356b41** [comment]
can you answer my questions? @metral

**Comment by metral** [comment]
> first one is intentional, i dont like it when its all at 1.

We do not use specific numbers, and intentionally want to mark them as `1.` This is to have markdown auto-generate the numbers for us, and in the event that this doc gets updated in the future, we don't want / have to re-create the numbering.

> 2-x. like I said, you want them to look the same, ok. that is easy.
> kubectl section isnt missing, its just a bit shorter (again, you want them to look exactly the same?)

Yes. We we want identical content.

> experimentation is not possible with python, its cant do yaml files yet.

Understood, then leave this out if it is not capable with python.

> not clear about summary. you want me to literally copy-paste it or something else?

Yes.

> not sure how to resolve that conflict, that change is intentional, since the original contains many errors.

You'll have to rebase your changes on the latest master and manually address your conflicts, or else a merge will not be possible.

**Comment by 4c74356b41** [comment]
```
We do not use specific numbers, and intentionally want to mark them as 1. This is to have markdown auto-generate the numbers for us, and in the event that this doc gets updated in the future, we don't want / have to re-create the numbering.
```
makes no sense, markdown will autonumber them doesnt matter what you put in there

image

**Comment by hausdorff** [comment]
I sense we are getting off track. :)

I think we're probably open to re-evaluating the (un-written-down) style guide, but that's a big topic and we should do that outside of this PR.

@4c74356b41 You are correct that the docs now contain many errors, but the style suggestions seem like a reasonable amount of work -- and in fact, I think @metral suggested them precisely _because_ they are easy. Are you willing to implement them? If not, it's best to make that explicit now, so that we can decide what to do.

**Comment by 4c74356b41** [comment]
reasonable ones - yes. "change 1,2,3,4,5 to 1,1,1,1,1" - no. this is unreasonable.

**Comment by hausdorff** [comment]
Ok, then let's fix the other things but leave the numbering, and we'll
merge.

One meta-point for all involved here. :) The point of a style is so that we
*don't* have argue about things like this! Many of those choices are
arbitrary or confer only minor technical benefit, and many times the
changes seem trivial. But the benefit is that we *don't* spend days
debating how to number lists.

**Comment by 4c74356b41** [comment]
build failed because of the other PR of mine not accepted to the examples repo (not even a response in 2 weeks).

**Comment by 4c74356b41** [comment]
ok, I'll submit a new one for the kubernetes bit this week.

**Pull request merged**